packages/iov-bcp-types/src/atomicswapmerger.ts

```diff
@@ -0,0 +1,73 @@
+import { Encoding } from "@iov/encoding";
+
+import { AtomicSwap, OpenSwap, SwapState } from "./atomicswap";
+import { SwapClaimTransaction, SwapTimeoutTransaction } from "./transactions";
+
+function settleAtomicSwap(swap: OpenSwap, tx: SwapClaimTransaction | SwapTimeoutTransaction): AtomicSwap {
+  if (tx.kind === "bcp/swap_claim") {
+    return {
+      kind: SwapState.Claimed,
+      data: swap.data,
+      preimage: tx.preimage,
+    };
+  } else {
+    return {
+      kind: SwapState.Expired,
+      data: swap.data,
+    };
+  }
+}
+
+export class AtomicSwapMerger {
```

Comment:
Nice, so this takes a series of updates/transactions and locally materializes a state from it? I like it

Reply:
Jap, more or less what you did inline before but extracted in a class dedicated to that job.